src/frontends/tensorflow/src/op/concat.cpp

```diff
@@ -34,17 +34,47 @@ OutputVector translate_concat_op(const NodeContext& node) {
     get_const_input(node, axis_idx, &tf_concat_axis_vec);
     int64_t concat_axis = tf_concat_axis_vec[0];
 
+    // Excluding zero-dim inputs
     OutputVector ng_args;
     for (int i = concat_idx_start; i < concat_idx_stop; i++) {
         Output<Node> ng_arg = node.get_input(i);
-        ng_args.push_back(ng_arg);
-    }
+        bool valid_input = true;
 
-    auto res = make_shared<Concat>(ng_args, size_t(concat_axis));
-    set_node_name(node.get_name(), res);
-    return res->outputs();
+        if (ng_arg.get_partial_shape().is_static()) {
+          auto inp_shape = ng_arg.get_shape();
+          for (auto dim : inp_shape) {
+            if (dim == 0) {
+              valid_input = false;
+              break;
+            }
+          }
+        }
```

Comment:
you can apply this not only to static shapes but it also regards partial shapes (with dynamic dimensions). Just check that rank is static `arg.get_partial_shape().rank().is_static()` and dimension is static `arg.get_partial_shape()[dim].is_static()`

Reply:
Done